src/console_io/include/window.hpp

```diff
@@ -0,0 +1,44 @@
+#ifndef __WINDOW_HPP
+#define __WINDOW_HPP
+
+#include <memory>
+#include <ncurses.h>
+#include <string>
+#include <vector>
+
+namespace console_io
+{
+class window
+{
+      protected:
+        WINDOW *win_ptr = nullptr;
+
+      public:
+        int size_x     = 0;
```

Comment:
It is impossible to change window size using Curses. It could be done only by creating a new window and deleting the old one (that will end up in recursive copying of subwindows).